multiselect/add-new-item/MultiSelectAndAddButton/MultiSelectAndAddButton/Shared/MultiSelectWithAdd.razor

```diff
@@ -50,17 +61,31 @@
     async Task OnCreateRecord()
     {
         OptionsModel insertedOption = await _dataService.AddOption(_currentFilterValue);
-        Options.Add(insertedOption); // update the current view-model. Add() works because it is an observable collection
 
-        // update the values, needs a new reference, see https://docs.telerik.com/blazor-ui/common-features/observable-data#refresh-data
-        // continues with a hack in OnRead
+        // continues with a hack in OnRead and OnAfterRenderAsync
         Tags.Add(insertedOption.MyValueField);
-        shouldUpdateValueParam = true;
+        shouldClearInput = true;
+
+        MultiSelectWithButton.Rebind();
 
         // raises event for parent componets to use
         await TagsChanged.InvokeAsync(Tags);
     }
 
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (shouldUpdateTags)
+        {
+            shouldUpdateTags = false;
+
+            // hack to refresh the selected values AFTER the MultiSelect is rebound
+            Tags = new List<int>(Tags);
```

Comment:
This is needed, because int is not ref type. We can skip this hack if we change the Value type of the multiselect.

For example we can just wrap the int in one object:
```csharp
public class MultiselectModel
{
    public int Value { get; set; }
}
```

and when add value we can call: `Tags.Add(new MultiselectModel { Value = insertedOption.MyValueField });`

Reply:
`Tags` is not the MultiSelect data, but the list of selected values. Shouldn't `TValue` and `ValueField` be always of primitive type? Also, if this approach is possible, it will be inconsistent with all our docs and demos. If the customer switches back to a List<int>, the example will break.

Reply:
Looks like I totally misunderstood the example. Sorry about that. Disregard my comments.